src/Xamarin.Android.Build.Tasks/Tasks/MavenDownloadTask.cs

```diff
@@ -0,0 +1,207 @@
+			var id = library.ItemSpec;
+			var version = library.GetRequiredMetadata ("Version", Log);
+
+			if (version is null)
+				continue;
+			
+			var artifact = MavenExtensions.ParseArtifact (id, version, Log);
+			
+			if (artifact is null)
+				continue;
+
+			// Check for local files
+			if (GetLocalArtifactOrDefault (library, Log) is TaskItem cached_result) {
+				library.CopyMetadataTo (cached_result);
+				resolved.Add (cached_result);
+				continue;
+			}
+
+			// Check for repository files
+			if (await GetRepositoryArtifactOrDefault (artifact, library, Log) is TaskItem result) {
+				library.CopyMetadataTo (result);
+				resolved.Add (result);
+				continue;
+			}
+		}
+
+		ResolvedAndroidMavenLibraries = resolved.ToArray ();
+	}
+
+	TaskItem? GetLocalArtifactOrDefault (ITaskItem item, TaskLoggingHelper log)
+	{
+		// Handles a Repository="file" entry, like:
+		//  <AndroidMavenLibrary 
+		//    Include="my.company:package" 
+		//    Version="1.0.0" 
+		//    Repository="File"
+		//    PackageFile="C:\packages\mypackage-1.0.0.jar"
+		//    PomFile="C:\packages\mypackage-1.0.0.pom" />
+		var type = item.GetMetadataOrDefault ("Repository", "Central");
+
+		if (type.Equals ("file", StringComparison.InvariantCultureIgnoreCase)) {
+			var artifact_file = item.GetMetadataOrDefault ("PackageFile", string.Empty);
+			var pom_file = item.GetMetadataOrDefault ("PomFile", string.Empty);
+
+			if (!artifact_file.HasValue () || !pom_file.HasValue ()) {
+				log.LogError ("'PackageFile' and 'PomFile' must be specified when using a 'File' repository.");
+				return null;
+			}
+
+			if (!File.Exists (artifact_file)) {
+				log.LogError ("Specified package file '{0}' does not exist.", artifact_file);
+				return null;
+			}
+
+			if (!File.Exists (pom_file)) {
+				log.LogError ("Specified pom file '{0}' does not exist.", pom_file);
+				return null;
+			}
+
+			var result = new TaskItem (artifact_file);
+
+			result.SetMetadata ("ArtifactSpec", item.ItemSpec);
+			result.SetMetadata ("ArtifactFile", artifact_file);
+			result.SetMetadata ("ArtifactPom", pom_file);
+
+			return result;
+		}
+
+		return null;
+	}
+
+	async System.Threading.Tasks.Task<TaskItem?> GetRepositoryArtifactOrDefault (Artifact artifact, ITaskItem item, TaskLoggingHelper log)
+	{
+		// Handles a Repository="Central|Google|<url>" entry, like:
+		//  <AndroidMavenLibrary 
+		//    Include="androidx.core:core" 
+		//    Version="1.9.0" 
+		//    Repository="Google" />
+		// Note if Repository is not specifed, it is defaulted to "Central"
+
+		// Initialize repo
+		var repository = GetRepository (item);
+
+		if (repository is null)
+			return null;
+
+		artifact.SetRepository (repository);
+
+		// Download artifact
+		var artifact_file = await MavenExtensions.DownloadPayload (artifact, MavenCacheDirectory, Log);
+
+		if (artifact_file is null)
+			return null;
+
+		// Download POM
+		var pom_file = await MavenExtensions.DownloadPom (artifact, MavenCacheDirectory, Log);
```

Comment:
Do these `MavenExtensions` methods have an option to pass in a `CancellationToken`? If not, we might have to go add them.

`AndroidAsyncTask` implements MSBuild's `ICancelableTask`, and there is a `CancellationToken` property. If we don't pass this in, if you hit Ctrl+C during a build it won't be able to cancel these async calls. Similar behavior if you hit the `Cancel Build` button in VS.